homeassistant/components/generic_thermostat/climate.py

```diff
@@ -521,13 +518,17 @@ async def _async_heater_turn_off(self):
 
     async def async_set_preset_mode(self, preset_mode: str):
         """Set new preset mode."""
-        if preset_mode == PRESET_AWAY and not self._is_away:
-            self._is_away = True
+        if preset_mode not in SUPPORTED_PRESETS:
+            raise ValueError(
+                f"Got unsupported preset_mode {preset_mode}. Must be one of {SUPPORTED_PRESETS}"
+            )
+        if preset_mode == PRESET_AWAY and preset_mode != self._attr_preset_mode:
```

Comment:
Why not make another if-check to see if preset == current preset and ignore ? 

```python
if preset_mode == self._attr_preset_mode:
    pass
```

However, should we actually not call it? Or would it be to the to-be-called service to decide it should ignore it.